torch/distributed/_tensor/placement_types.py

```diff
@@ -54,37 +54,102 @@ def _split_tensor(
             tensor.size(self.dim) >= num_chunks
         ), f"Tensors to be sharded on dim {self.dim} must be at least as large as "
         f"the number of devices in that dimension {num_chunks}"
-        # split tensor over dimension `dim` into n slices with padding if necessary
-        tensor_list = list(tensor.tensor_split(num_chunks, self.dim))
-        idx_start_to_pad = tensor.size(self.dim) % num_chunks
-        if with_padding or contiguous:
+
+        # chunk tensor over dimension `dim` into n slices with padding if necessary
+        tensor_list = list(torch.chunk(tensor, num_chunks, dim=self.dim))
+        # compute the chunk size inline with ``torch.chunk``
+        full_chunk_size = (tensor.size(self.dim) + num_chunks - 1) // num_chunks
+
+        # Compute chunk size for each chunk for ``self.dim``
+        chunk_sizes = [
+            tensor_list[idx].size(self.dim) if idx < len(tensor_list) else 0
+            for idx in range(num_chunks)
+        ]
+        # Get idx start to pad
+        idx_start_to_pad = next(
+            (
+                idx
+                for idx, _ in enumerate(chunk_sizes)
+                if idx > 0 and chunk_sizes[idx] < chunk_sizes[idx - 1]
+            ),
+            0,
+        )
+        # Compute pad size on each chunk
+        pad_sizes = [
+            full_chunk_size - chunk_size if idx >= idx_start_to_pad else 0
+            for idx, chunk_size in enumerate(chunk_sizes)
```

Comment:
i.e. we don't really need to check `if idx >=idx_start_to_pad`? it could always be `full_chunk_size-chunk_size`, for ranks don't need to pad, the subtraction would become 0 automatically?